lectures/12-plonk.tex

```diff
@@ -0,0 +1,234 @@
+\end{enumerate}
+
+This would imply that \(\hat{f}(X,Y) = \hat{g}(X,Y)\) with high probability due to Schwartz-Zippel.
+
+\begin{remark}
+Complete and sound, assuming \(2d/p\) is negligible.
+\end{remark}
+
+\subsection{Plonk Arithmetization}
+
+Assume that we have a certain arithmetic circuit \(C\) with a \(|C|\) number of gates and \(|I|\ = |I_x| + |I_w|\) number of inputs. We encode this circuit, recording its computation trace in a table, where rows represent the state per each gate, while columns are of the form \((a, b, c)\), where \(a\) and \(b\) are left and right inputs, and \(c\) is the output of the gate. In this manner, the output of the last gate corresponds to the output of the circuit.
+
+\begin{example}
+Consider this circuit: \((x_1 + x_2)*(x_1 + w_1)\). Suppose we set \(x_1 = 5, x_2 = 6, w_1 = 1\). Then the computation trace would be following:
+
+\begin{tabular}{l rrr}
+  inputs: & 5, & 6, & 1 \\ \hline
+  Gate 0: & 5, & 6, & 11 \\
+  Gate 1: & 6, & 1, & 7 \\
+  Gate 2: & 11, & 7, & 77 \\ 
+\end{tabular}
+
+\end{example}
+
+\subsubsection{Encoding the trace as a polynomial}
+
+Let \(d = 3|C| + |I|\) and \(\Omega = \{1, \omega^1, \omega^2, \dots, \omega^{d-1}\}\). Then we would like to interpolate a polynomial \(T \in \mathbb{F}_p^{(\leq d)}[X]\) to encode the entire computation trace in a succinct, processing-prone form.
+
+\begin{enumerate}
+    \item \(T\) \textbf{encodes all inputs:} \( T(\omega^{-j}) = \text{input \#} j \text{for } j = 1, \dots, |I| \)
+    \item \(T\) \textbf{encodes all wires:} \( \forall \ell = 0, \dots, |C| - 1: \)
+    \begin{itemize}
+        \item \( T(\omega^{3\ell}) \): left input to gate \#\( \ell \)
+        \item \( T(\omega^{3\ell+1}) \): right input to gate \#\( \ell \)
+        \item \( T(\omega^{3\ell+2}) \): output of gate \#\( \ell \)
+    \end{itemize}
+\end{enumerate}
+
+\begin{remark}
+In this way, we obtain the polynomial \(T\) in evaluation form. It is possible to compute coefficients of \(T\) using FFT in \(O(d\log(d))\).
+\end{remark}
+
+\begin{example}
+For our example circuit, we have \(|C| = 3\) and \(|I| = 3\), therefore \(degree(T) = 11\). The prover interpolates the polynomial \(T(X)\) such that:
+\\
+\begin{tabular}{l l l l}
+  inputs: & $T(\omega^{-1}) = 5$, & $T(\omega^{-2}) = 6$, & $T(\omega^{-3}) = 1$, \\ \hline
+  gate 0: & $T(\omega^{0}) = 5$, & $T(\omega^{1}) = 6$, & $T(\omega^{2}) = 11$, \\
+  gate 1: & $T(\omega^{3}) = 6$, & $T(\omega^{4}) = 1$, & $T(\omega^{5}) = 7$, \\
+  gate 2: & $T(\omega^{6}) = 11$, & $T(\omega^{7}) = 7$, & $T(\omega^{8}) = 77$ \\ 
+\end{tabular}
+
+\end{example}
+
+\subsection{Proving the validity of \texorpdfstring{\(T\)}{T}}
+
+After prover \(P(S_p, x, w)\) has constructed \(T\), it commits it and sends to the verifier \(V(S_v, x)\). Latter must verify (meaning former must prove) four points about the validity of constructed \(T\), which we will describe in next sections.
```

Comment:
We used $\mathcal{P}$ to denote the prover. Prover parameters are denoted by $\mathsf{pp}$, while verifier parameters by $\mathsf{vp}$. Just to that we keep the notation consistent.